### ISSUE TITLE
Introduce a functional option for `span.RecordError` to request that sets the status if the supplied `error` is not nil

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#1677]
_Originally posted by @seh in https://github.com/open-telemetry/opentelemetry-go/issues/1661#issuecomment-791762744_

**Comment by lastchiliarch** [comment]
I'd like to take this one.

I'm going to add an  IsRecordError filed to SpanConfig and new event option WithRecordError.
If spanConfig.IsRecordError is true when RecordError fucntion is called with options inlucde  WithRecordError, I will call the s.SetStatus(codes.Error, "") to set status. 

Glad to hear from you, if anything was misunderstood.

**Comment by MrAlias** [comment]
> I'm going to add an IsRecordError filed to SpanConfig and new event option WithRecordError.
> If spanConfig.IsRecordError is true when RecordError fucntion is called with options inlucde WithRecordError, I will call the s.SetStatus(codes.Error, "") to set status.

I'm not sure this is what I or others were thinking originally. Instead, add [an EventOption](https://github.com/open-telemetry/opentelemetry-go/blob/d616df61f5d163589228c5ff3be4aa5415f5a884/trace/config.go#L103) that was named `WithStatus`. E.g.

```
func WithStatus() EventOption
```

 This would be passed to the call of `RecordError` when instrumentation wanted to record an error and set the span status to `Error`.

**Comment by treuherz** [comment]
It looks like this got stuck after #1677 fell behind a bunch of refactors. If @lastchiliarch isn't able to take it forward I'd be interested in picking this up in a new PR.

**Comment by tmc** [comment]
@treuherz this would be nice to see in.

**Comment by treuherz** [comment]
Apologies, I wasn’t able to find a satisfactory backwards-compatible way of making this happen. Maybe inspiration will strike someone else!

**Comment by amanakin** [comment]
Hey @MrAlias, tried to solve an issue.
Can you take a look at PR, please?

**Comment by pellared** [comment]
Closing as we are planning to deprecate `span.RecordError`. 
References: 
- https://github.com/open-telemetry/opentelemetry-specification/pull/4825 
- https://opentelemetry.io/blog/2026/deprecating-span-events/

However, I created https://github.com/open-telemetry/opentelemetry-specification/issues/5005 as an alternative approach.
